openerp/addons/base/ir/ir_model.py

```diff
@@ -431,6 +435,19 @@ def _drop_column(self, cr, uid, ids, context=None):
 
         return True
 
+    @api.multi
+    def _prepare_update(self):
+        """ Check whether the fields in ``self`` may be modified or removed.
+            This method prevents the modification/deletion of many2one fields
+            that have an inverse one2many, for instance.
+        """
+        for record in self:
+            model = self.env[record.model]
```

Comment:
@rco-odoo What about if we want to delete a `ir.model` record where a module is not exists anymore?
I mean, If a `model` is not registered in the environment.

We have a `KeyError: u'mode.name'` with the following traceback deleting a model with this case:
```bash
Traceback (most recent call last):
`   File "~/build/OCA/server-tools/database_cleanup/tests/test_database_cleanup.py", line 65, in test_database_cleanup
...
`   File "~/odoo/openerp/addons/base/ir/ir_model.py", line 147, in unlink
`     model.field_id._prepare_update()
`   File "~/odoo/openerp/addons/base/ir/ir_model.py", line 445, in _prepare_update
`     model = self.env[record.model]
`   File "~/odoo/openerp/api.py", line 768, in __getitem__
`     return self.registry[model_name]._browse(self, ())
`   File "~/odoo/openerp/modules/registry.py", line 84, in __getitem__
`     return self.models[model_name]
` KeyError: u'x_database.cleanup.test.model'
```

cc @dbo-odoo @odony @KangOl @hbrunn 
Related https://github.com/OCA/server-tools/pull/612#issuecomment-261526850